lib/matplotlib/figure.py

```diff
@@ -2747,41 +2800,14 @@ def set_figheight(self, val, forward=True):
         """
         self.set_size_inches(self.get_figwidth(), val, forward=forward)
 
-    def clf(self, keep_observers=False):
-        """
-        Clear the figure.
-
-        Set *keep_observers* to True if, for example,
-        a gui widget is tracking the Axes in the figure.
-        """
-        self.suppressComposite = None
-        self.callbacks = cbook.CallbackRegistry()
-
-        for ax in tuple(self.axes):  # Iterate over the copy.
-            ax.cla()
-            self.delaxes(ax)         # removes ax from self._axstack
-
-        toolbar = getattr(self.canvas, 'toolbar', None)
+    def clear(self, keep_observers=False):
+        # docstring inherited
+        super().clear(keep_observers=keep_observers)
+        # FigureBase.clear does not clear toolbars, as
+        # only Figure can have toolbars
+        toolbar = self.canvas.toolbar
         if toolbar is not None:
             toolbar.update()
-        self._axstack.clear()
```

Comment:
This is different than `main` where it replaced it with a completely new object; I wonder if we still need to clear it on the branch?

Reply:
no, I think this was always redundent.  In `delaxes` we remove the axes from the stack (which internally calls `self.clear()` and then re-adds everything _but_ the removed elmenet).